### ISSUE TITLE
Update notifications

### ISSUE THREAD

**Issue by M-Bab** [M-Bab/linux-kernel-amdgpu-binaries#23]
This isn't a real issue. It is just there to inform interested people about pushed updates. I will add an entry on each pushed major update.

Please do not comment to this issue.

**Comment by M-Bab** [comment]
Updated to kernels 4.9.50+, 4.12.13+, 4.12.13+_UBUNTU
Updated firmware package with Raven binaries.
Updated README.

**Comment by M-Bab** [comment]
Updated to kernels 4.9.51+, 4.12.14+, 4.12.14+_UBUNTU

**Comment by M-Bab** [comment]
Updated to kernels 4.9.53+, 4.13.5+, 4.13.5+_UBUNTU

**Comment by M-Bab** [comment]
Updated to kernels 4.9.55+, 4.13.6+, 4.13.6+_UBUNTU

**Comment by M-Bab** [comment]
Updated to kernels 4.9.59+, 4.13.10+, 4.13.10+_UBUNTU
Updated firmware binaries.

**Comment by M-Bab** [comment]
Updated to kernels 4.14.0+ and 4.14.0+_Ubuntu
Kernel 4.9 is removed because 4.14 is the next longterm support (LTS) kernel.
Caution: Issues when re-activating the screen from standby or after screen lock can still occur. I have kind of given up on the issue - I could find nothing that helps the AMD developers to resolve the issue.

**Comment by M-Bab** [comment]
Updated to kernels 4.15+ and 4.15+_Ubuntu
These are pre-release kernels but amd-staging-drm-next jumped to 4.15 so I had to follow. At least it should resolve the standby issues.

**Comment by M-Bab** [comment]
Updated to kernels 4.15.0-rc4+, 4.15.0-rc4+_UBUNTU

**Comment by M-Bab** [comment]
Updated to kernels 4.15.0-rc5+, 4.15.0-rc5+_UBUNTU

**Comment by M-Bab** [comment]
Updated to kernels 4.15.0-rc6+, 4.15.0-rc6+_UBUNTU
Happy New Year!

**Comment by ghost** [comment]
Thanks Martin. Running it now without issue on Mint 18.3, Compiz enabled.
Amd Radeon R6 Carrizo w/ Vulkan. Thanks for your work, and Happy New Year!
​
​
​

**Comment by M-Bab** [comment]
Updated to kernels 4.15.0-rc8+, 4.15.0-rc8+_UBUNTU

**Comment by M-Bab** [comment]
Updated to kernels 4.15.1+, 4.15.1+_UBUNTU

Little delay I was pretty busy. So 4.15 is officially out and includes DAL hooray! Now you can also get your kernel with sound from more official sources like: http://kernel.ubuntu.com/~kernel-ppa/mainline/?C=N;O=D

Why did I still build this one and why it might (depends on the case) still be worth to install it?
1. This kernel has the Ubuntu additions from unstable in contrast to the e.g. mainline kernel.
2. The amdgpu code is way newer than in 4.15. Concerning the amd driver it is rather a 4.16 kernel already. DAL version 3.1.32 vs 3.1.07.

**Comment by M-Bab** [comment]
Updated to kernels 4.15.4+, 4.15.4+_UBUNTU

**Comment by M-Bab** [comment]
Kernel 4.16 based on amd-staging-drm-next (vanilla and Ubuntu) replaces 4.15.
    VBlank crash after screen lock is finally fixed. Screen locking can still result in an unresponsive system or screen flickering.

**Comment by M-Bab** [comment]
Updated to kernels 4.16.0-rc7+, 4.16.0-rc7+_UBUNTU

**Comment by M-Bab** [comment]
Updated to kernels 4.16.0, 4.16.0_UBUNTU Release versions.
Updated firmware package (Vega12, Vega20).

**Comment by M-Bab** [comment]
Updated to kernels 4.16.2+, 4.16.2+_UBUNTU
Flickering not fixed unfortunately.

**Comment by M-Bab** [comment]
Rebuild 4.16.2+ just to get rid of the flickering.

**Comment by M-Bab** [comment]
Updated to kernels 4.16.3+, 4.16.3+_UBUNTU

**Comment by M-Bab** [comment]
Updated to kernel 4.16.15+

**Comment by M-Bab** [comment]
Kernel 4.18 based on amd-staging-drm-next (vanilla and Ubuntu) replaces 4.17.

**Comment by M-Bab** [comment]
Updated to kernel 4.18.0 RELEASE

**Comment by M-Bab** [comment]
Updated to kernel 4.18.5

**Comment by M-Bab** [comment]
Updated to kernel 4.18.8, updated firmware binaries.

**Comment by M-Bab** [comment]
Updated to kernel 4.19.1

**Comment by M-Bab** [comment]
After a long break: Updated to kernel 5.0-rc4

**Comment by AndreaBorgia-Abo** [comment]
5.0-rc6 was pushed by Martin 12 days ago, self-notifying after the fact :)

**Comment by M-Bab** [comment]
Updated to kernel 5.0-rc8.

**Comment by M-Bab** [comment]
Updated to kernel 5.0, updated firmware binaries package.

**Comment by M-Bab** [comment]
Updated to kernel 5.0.1. Is there still interest in such a bleeding edge kernel? Ubuntu 19.04 will include the 5.0 kernel most likely with freesync and I am not sure if this project needs to be continued.

**Comment by AndreaBorgia-Abo** [comment]
@M-Bab yes, very much so, thank you! 
I am using Debian / testing for my Ryzen 3 2200G and this kernel is a real treasure :)

**Comment by lillelilje** [comment]
This project is a godsend for those of us who likes experimenting with "bleeding edge" hardware, so I'd personally love to see the project continued.

**Comment by M-Bab** [comment]
Updated to kernel 5.0.2.

**Comment by AndreaBorgia-Abo** [comment]
Update to kernel 5.0.4

**Comment by M-Bab** [comment]
Updated kernels to 5.0.9. Updated firmware binaries.

**Comment by AndreaBorgia-Abo** [comment]
Updated kernels to 5.0.17. Updated firmware binaries.
(addendum: no, I did not do it... I just noticed there is a new kernel and wrote a comment here :D )

**Comment by DarkFoss** [comment]
I just wanted to say Thank you for continuing to roll these kernels. I'm still on Kde-Neon and your kernels are still the best way to have freesync + the latest cve protection. For what ever reason Ubuntu mainline kernels have dropped the freesync patch after 5.1_rc2. None of them have the option to turn on freesync, nor does newest 5.2_rc1 that includes the drm-tip and daily branches.
Sorry for not filling out or adding to bug reports, I've been busy over the last few months trying to track down [drm] dce110_link_encoder_construct: Failed to get encoder_cap_info from VBIOS with error code 4! using the mainline kernels turns out it started with 4.18_rc1 which means it's time to learn to bisect and go through 120+ AMD commits.

**Comment by M-Bab** [comment]
Update to kernel 5.0.19.

**Comment by M-Bab** [comment]
Update to new kernel 5.2.0-rc6. Already includes Navi 10 support. If anyone buys such a card in the next weeks please inform me if it works!

**Comment by M-Bab** [comment]
Updated to released kernel 5.2.0.

**Comment by AndreaBorgia-Abo** [comment]
Updated to kernel 5.2.4 by @M-Bab

**Comment by AndreaBorgia-Abo** [comment]
Updated to kernel 5.2.5 ( @M-Bab )

**Comment by M-Bab** [comment]
Updated to kernel 5.2.6

**Comment by M-Bab** [comment]
Updated to kernel 5.2.9

**Comment by M-Bab** [comment]
Had to replace kernel 5.2.9 with another build. The original 5.2.9 crashed on boot on my PC.

**Comment by M-Bab** [comment]
Updated to kernel 5.2.10

**Comment by stuaxo** [comment]
Arg, after a having some trouble for a few kernel versions, having to boot to recovery mode, I've found nomodeset means I can boot... though brightness controls have stopped working.

Not really the right place to put this, but wish AMD and / or HP would get it together, I don't know if it's the kernel / firmware or the BIOS update / tweaking GRUB that's done it :/

**Comment by Mr-Precise** [comment]
Are updates planned?
You can give ready-made instructions or script for building the linux kernel + amd-staging.
And I might be able to release updates more often...
thanks

**Comment by M-Bab** [comment]
Yeah there are still updates planned and this on a weekly basis again. But to be honest: the builds on 5.3 were simply not successful. So I skipped them hoping that AMD will soon rebase to 5.4. If it helps I can still do the patch releases of 5.2 but for the other stuff I need properly building kernel code. I really hope it works well again with 5.4.

**Comment by M-Bab** [comment]
Okay tried it again with 5.3 - whatever changed in the meantime. Both kernels build again. Enjoy 5.3.11 based on amd-staging-drm-next.

**Comment by M-Bab** [comment]
Updated to kernel 4.2 with updated firmware binaries.

**Comment by stuaxo** [comment]
Updated, but I don't seem to have WIFI when I boot onto this kernel :/

**Comment by M-Bab** [comment]
Updated to kernel 5.4.8.

**Comment by JoneKone** [comment]
Oh man, after i bought Vega56 for me and my kid, this kernel has been the only thing that kept me sane, the computers just stopped crashing randomly thank you. (Black screen of death)

**Comment by M-Bab** [comment]
Updated to kernel 5.6.12 (debian+ubuntu variants). Updated firmware binaries package.

@JoneKone is the "oh man" an expression of concern this kernel died? It didn't - I just did not have a lot time recently. Another factor is that may graphics card is so old in the meantime it does not ultimately need this kernel. This was different when I started to build it (no HDMI sound and stuff).

**Comment by JoneKone** [comment]
@M-Bab It was a sigh of relief =) Vega 56 isnt' that old, and a guy on Linux gaming group telegram, is having so many issues with rx 5700xt... Similar issues that i'm having.  So thankyou :)

**Comment by M-Bab** [comment]
Updated kernel to 5.6.19, updated firmware binaries. Kernel already includes code for upcoming sienna but the firmware binaries are not there yet.

**Comment by M-Bab** [comment]
Updated to kernel 5.9.0 **-rc5**. Happily recognized AMD rebased their kernel from 5.6 to a newer one. Also nice: Code for future graphic cards (sienna cichlid, navy flounder) already included. I will look out for the matching firmware binaries.

**Comment by M-Bab** [comment]
For the first time I also published it as release with assets: https://github.com/M-Bab/linux-kernel-amdgpu/releases/tag/linux-5.9.0-rc5-20.09.19.amdgpu . But not sure if this is more helpful.

**Comment by JoneKone** [comment]
https://www.phoronix.com/scan.php?page=news_item&px=Linux-5.9-rc6-Released   Released With Soft Scrollback Removed, Performance Regression Fixed.

Hmm

**Comment by stuaxo** [comment]
I recently got an HP ds000xxx with a Ryzen 3700u and it's really broken as in ACPI they set "reduced hardware" to 1, so it doesn't boot by default [1].

Any idea if it will work any better with 5.9 ?

(I believe there was a bug where Linux didn't handle the "reduced hardware" setting very well)

[1 ]https://askubuntu.com/questions/1162746/why-wont-my-hp-envy-x360-laptop-with-ryzen-7-3700u-boot-ubuntu-live-cd

**Comment by M-Bab** [comment]
Updated to kernel 5.9.0 **-rc6/-rc7**. The last one wasn't a very good one unfortunately. On the one side the performance regression mentioned above and - at least for me - hibernation did not work. Both should be fixed now.

**Comment by lanigb** [comment]
Hello,
thanks for your work.
I installed the latest kernel you compiled (5.9.0-rc8-20.10.05.amdgpu.ubuntu) but was not able to get out of hibernation.

```
Oct 24 09:12:08 logodenn kernel: [  206.513174] amdgpu 0000:38:00.0: [drm:amdgpu_ring_test_helper [amdgpu]] *ERROR* ring gfx test failed (-110)
Oct 24 09:12:08 logodenn kernel: [  206.513210] [drm:amdgpu_device_ip_resume_phase2 [amdgpu]] *ERROR* resume of IP block <gfx_v9_0> failed -110
Oct 24 09:12:08 logodenn kernel: [  206.513212] amdgpu 0000:38:00.0: amdgpu: amdgpu_device_ip_resume failed (-110).
Oct 24 09:12:08 logodenn kernel: [  206.513216] PM: dpm_run_callback(): pci_pm_restore+0x0/0x90 returns -110
Oct 24 09:12:08 logodenn kernel: [  206.513223] PM: Device 0000:38:00.0 failed to restore async: error -110
```
I didn't have the amdgpu.dc=1 set in the boot parameters then.
Do you understand the reason and would you know a fix ?

Moreover, syslog is full of  messages like
 `evbug: Event. Dev: input2, Type: 1, Code: 103, Value: 1`

Do this messages relate to a real problem or may I safely blacklist the module?

TIA

**Comment by M-Bab** [comment]
Updated to kernel 5.9.1.

**Comment by M-Bab** [comment]
@lanigb Please first try the fresh 5.9.1 - there **were** some issues with hibernation in the past week. If it is not fixed please open with error log as a new issue.

**Comment by lanigb** [comment]
Hi, 
Came back home and tried the new kernel. It is actually worst. What log should I post ? syslog ? I don't have an error log.

**Comment by JoneKone** [comment]
Any news on getting 5.10.1 working?

**Comment by M-Bab** [comment]
@JoneKone Unfortunately amd-staging-drm-next has still not been rebased. I will precisely start to build kernels again once the first lines of this file: https://cgit.freedesktop.org/~agd5f/linux/tree/Makefile?h=amd-staging-drm-next show it has been lifted on any newer kernel than 5.9.

For know the deviation to the official sources of 5.9 are too big so it would be really effortful to get it merged and building (not even to mention working!) again. This also increases the risks of crashes and so on - so I would like to avoid that. So far the AMD guys were pretty fast jumping on the most recent kernel (except they stuck really long with 4.8 - but thats a while ago :smile: ).

**Comment by JoneKone** [comment]
:)

**Comment by stuaxo** [comment]
Can us AMD users crowdsource an AMD focused kernel dev ?    I don't see AMD doing enough to fix things that have been broken for years in Ryzen.

**Comment by M-Bab** [comment]
No idea. Usually the AMD open source team was pretty strong and active. No idea what is happening now. So far their usual repository feels somehow abandoned since almost 2 months.

I hope they will pick it up again.

**Comment by stuaxo** [comment]
Just airing a bit of frustration, certainly things are a lot better than they were.. a lot of my issues seem to be down to HPs broken bios.  I have a laptop with a Ryzen 2500u that is pretty OK (but still with the odd freeze here and there).   One with a 3700u that is unused rn because things were such a mess (very very broken bios), that I'm leaving it for a while to see if some of the fixes for the broken bios end up upstream.

**Comment by JoneKone** [comment]
Nice :) thx

**Comment by M-Bab** [comment]
Finally after quite a while I simply asked the AMD developers and they actually moved their repo (to https://gitlab.freedesktop.org/agd5f/linux/-/tree/amd-staging-drm-next ). The other repo was indeed abandoned.

So here we go. Kernel 5.11 based amd-staging-drm-next. The firmware files are also really outdated - maybe on the weekend I have time to collect all the neat little blobs together. As always: Use at your own risk.

**Comment by stuaxo** [comment]
Oh wow, fantastic news :)

**Comment by M-Bab** [comment]
Updated to kernel 5.11.8. Updated firmware binaries.

**Comment by M-Bab** [comment]
Updated to kernel 5.11.15. Updated firmware binaries.

Completely started from scratch because the previous kernel had graphic artifacts. Testing this one so far it looks fixed. Also updated the firmware binaries with ROCM blobs - so there are less warnings when this kernel is installed. And maybe someone actually wants to use this for open compute stuff.

**Comment by M-Bab** [comment]
Updated to kernel 5.11.18.

**Comment by stuaxo** [comment]
Alright, it's not out yet, but looks like 5.15 will finally do temperature reporting for Zen 2 / 3 APUs via k10temp with much qudos to Mario Limonciello.

https://www.phoronix.com/scan.php?page=news_item&px=Linux-5.15-HWMON

**Comment by M-Bab** [comment]
Its been a while ...
Updated to kernel 5.13.18

@stuaxo I assume, although the kernel provides the newest AMD *GPU* features, such *CPU* features will arrive as slowly as for all other kernels ...

**Comment by M-Bab** [comment]
Updated to kernel 5.13.19

5.13 is end-of-life but amd-staging-drm-next is still on this version.

**Comment by JoneKone** [comment]
seems to be updated to 5.16

**Comment by JoneKone** [comment]
I'm little bit afraid to ask this.. Are you still alive?

**Comment by M-Bab** [comment]
I am :smile: 

I did try to build 5.16 - but it did not build very well. Lots of merge conflicts - resolved them wrong it seems. Will try another round when I am back from vacation.

And also bad: Ubuntu abandoned their ubuntu-unstable repo where I could get all the Ubuntu modifications for each kernel versions. Now they only support the kernels they actually have in a distro (kind of makes sense, but bad for me). So probably the 5.16 kernel will not be available in Ubuntu flavor ...

**Comment by JoneKone** [comment]
:I uh oh, i have ubuntu :) have a nice vacation..

**Comment by stuaxo** [comment]
What a pain - onwards and upwards, hope the vacation is good.

**Comment by Mr-Precise** [comment]
I was able to build the 5.16 kernel, but it causes the graphics card to freeze for unknown reasons
(HD 7750)

**Comment by M-Bab** [comment]
## Update

My latest tries to build a mixed up kernel from the AMD code (and official linux kernel code and Ubuntu sauce) did not work well. There are massive merge conflicts and build errors which I dont have the time to figure out. But the reason for that is also kind of good news: The development in the official kernels is almost on-par with the latest AMD kernel developments.

I started this kernel because back in the days users didn't even have HDMI sound via AMD graphics card without such a special kernel. Things have improved a lot till then. The amdgpu dc code is merged for several years now and the software development there is so fast you usually get release day support for their new GPUs.

In short: This kernel is not really necessary anymore. I am not using it anymore either. I will shut these two repos down. Thanks to all users!

**Comment by themoonisacheese** [comment]
Nice job while you were there to help. 

I'm a bit surprised that you say support is now release day because I had to use this no earlier than September on a 5600G. That said, installing Ubuntu in place of Debian fixed it as well. Godspeed

**Comment by FlorianMittag** [comment]
Thanks for the work you put into this. Without your kernel my work notebook made all sorts of problems.

Godspeed and please know that some people are really grateful for your time and effort you put into this.

**Comment by JoneKone** [comment]
Thank you :)

**Comment by stuaxo** [comment]
Thanks for all the work over the years keeping two successive laptops of mine working, it has been much appreciated.

**Comment by JoneKone** [comment]
Well umm so, there is this weird project going on. https://forum.level1techs.com/t/mi25-stable-diffusions-100-hidden-beast/194172 aand couldn't help but notice that weirdly specific kernel versions seem to match. to your "old" project.

**Comment by stuaxo** [comment]
I know people used this repo for various different issues - one issue I had was my laptop not coming back from suspend.
amdgpu / kernel devs Mario Limonciello, Felix Kuehling and Aaron Liu helped me submit enough info that it will be mostly fixed in 6.4.

More info -

The issue:
On an HP x360 with Ryzen 3700u:  On returning from suspend the screen would be glitched or black, and I'd have to hard reset.

We found that fixes behaved differently depending on where the laptop gets power from (YMMV), the fix doesn't help if the laptop is powered by AC, but via USBC from a monitor or battery it now wakes up almost every time.

If you are having similar issues it may be worth building this kernel and testing.

Issue
https://gitlab.freedesktop.org/drm/amd/-/issues/2449

Quick thread
https://twitter.com/stu_axon/status/1648660362571464704

**Comment by M-Bab** [comment]
Pretty exciting there is still occasional news to this. And I am not even on Ubuntu anymore which means it would be even harder for me to build this kernel :sweat_smile:
